genie-web/src/main/java/com/netflix/genie/web/configs/GenieNotificationsAutoConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs;
+
+import com.netflix.genie.web.events.GenieEventBus;
+import com.netflix.genie.web.events.JobStateChangeObserverMulticaster;
+import com.netflix.genie.web.services.JobPersistenceService;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Beans to deliver job status changes notifications.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+public class GenieNotificationsAutoConfiguration {
+
+    /**
+     * Create a {@link JobStateChangeObserverMulticaster} and register it as observer of {@link JobPersistenceService}.
+     *
+     * @param genieEventBus         the genie event bus
+     * @param jobPersistenceService the job persistence service
+     * @return a {@link JobStateChangeObserverMulticaster} registered as observer of {@link JobPersistenceService}
+     */
+    @Bean
+    @ConditionalOnMissingBean(JobStateChangeObserverMulticaster.class)
+    public JobStateChangeObserverMulticaster jobStateChangeObserverMulticaster(
+        final GenieEventBus genieEventBus,
+        final JobPersistenceService jobPersistenceService
+    ) {
+        final JobStateChangeObserverMulticaster multicaster = new JobStateChangeObserverMulticaster(
+            genieEventBus
+        );
+        jobPersistenceService.addJobStateChangeObserver(multicaster);
```

Comment:
The reason i did it this way is that it forces this API to exist in the JobPersistence interface and so it has to be implemented by all implementations.

If i make it a constructor parameter, not every implementation will be obligated to support observers. And at the same time our particular implementation is "coupled" because it's doing things beyond what the interface requires.

Reply:
Yeah I just meant for your specific implementation you could pass it to the constructor as an initializer and still keep the API. Or in your bean creation just call the API in a loop before returning the bean. Either way it centralizes this instantiation rather than spreading it all over.

I'd also recommend a "remove" method to deregister listeners.